lib/node_modules/@stdlib/math/base/special/sici/src/main.c

```diff
@@ -0,0 +1,385 @@
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double polyval_gn8( const double x ) {
+	return 3.1404009894636335e-15 + (x * (3.859459254302766e-12 + (x * (1.7040445278204452e-9 + (x * (3.471311670841167e-7 + (x * (0.000034894116550227946 + (x * (0.001717182390523479 + (x * (0.03848787676499743 + (x * (0.33041097930563207 + (x * 0.6973599534432762)))))))))))))));
+}
+
+// END: polyval_gn8
+
+// BEGIN: polyval_gd8
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double polyval_gd8( const double x ) {
+	return 3.1404009894636335e-15 + (x * (3.878301660239547e-12 + (x * (1.7269374896631615e-9 + (x * (3.5704322344374083e-7 + (x * (0.00003684755044425611 + (x * (0.0019028442667439953 + (x * (0.04679131942596258 + (x * (0.48785225869530496 + (x * (1.6854889881101165 + (x * 1.0)))))))))))))))));
+}
+
+// END: polyval_gd8
+
+/* End auto-generated functions. */
+
+/**
+* Computes the sine and cosine integrals.
+*
+* ## Method
+*
+* -   The integrals are approximated by rational functions.
+*
+* -   For \\( x > 8 \\), auxiliary functions \\( f(x) \\) and \\( g(x) \\) are employed such that
+*
+*     ```tex
+*     \operatorname{Ci}(x) = f(x) \sin(x) - g(x) \cos(x) \\
+*     \operatorname{Si}(x) = \pi/2 - f(x) \cos(x) - g(x) \sin(x)
+*     ```
+*
+* ## Notes
+*
+* -   Absolute error on test interval \\( \[0,50\] \\), except relative when greater than \\( 1 \\):
+*
+*     | arithmetic | function    | # trials | peak    | rms     |
+*     |:----------:|:-----------:|:--------:|:-------:|:-------:|
+*     | IEEE       | Si          | 30000    | 4.4e-16 | 7.3e-17 |
+*     | IEEE       | Ci          | 30000    | 6.9e-16 | 5.1e-17 |
+*
+* @param x     input value
+* @param si    output pointer for sine integral
+* @param ci    output pointer for cosine integral
+*
+* @example
+* double si;
+* double ci;
+*
+* stdlib_base_sici( 3.0, &si, &ci );
+*/
+void stdlib_base_sici( const double x, double *si, double *ci ) {
+	double sgn;
+	double ax;
+	double c;
+	double f;
+	double g;
+	double s;
+	double z;
+
+	if ( stdlib_base_is_nan( x ) ) {
+		*si = 0.0 / 0.0; // NaN
+		*ci = 0.0 / 0.0; // NaN
+		return;
+	}
+
+	if ( x < 0.0 ) {
+		sgn = -1.0;
+		ax = -x;
+	} else {
+		sgn = 1.0;
+		ax = x;
+	}
+
+	if ( ax == 0.0 ) {
+		*si = 0.0;
+		*ci = STDLIB_CONSTANT_FLOAT64_NINF;
+		return;
+	}
+
+	if ( ax > 1.0e9 ) {
+		if ( stdlib_base_is_infinite( ax ) ) {
+			if ( sgn < 0.0 ) {
+				*si = -STDLIB_CONSTANT_FLOAT64_HALF_PI;
+				*ci = 0.0 / 0.0; // NaN
+			} else {
+				*si = STDLIB_CONSTANT_FLOAT64_HALF_PI;
+				*ci = 0.0;
+			}
+			return;
+		}
+		*si = STDLIB_CONSTANT_FLOAT64_HALF_PI - ( stdlib_base_cos( ax ) / ax );
```

Comment:
@anandkaranubc Here and below would be good opportunities to use `sincos`. I see that this also applies to the JS implementation.

Reply:
Hey @kgryte, thanks for pointing out this one. I wasn't aware that there is a function called `sincos` that actually calculates both `sin` and `cos` values simultaneously, removing the dependency of calling separate `sin` and `cos` functions. Hence, I have now incorporated that in my `sici` function's C implementation.